xapian-applications/omega/postprocess.py

```diff
@@ -0,0 +1,118 @@
+# Postprocess click data files for obtaining the final clickstream log file
+# and generate Query file for Xapian Letor module from it.
+#
+# Copyright (C) 2017 Vivek Pal
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+# USA
+
+from ast import literal_eval
+from sys import argv
+
+import csv
+
+script_name, search_log, clicks_log = argv
+
+try:
+    with open(search_log, 'r') as s:
+        with open('log/clickstream.log', 'w+') as f:
```

Comment:
For the two that have two nested `with` the combined line length doesn't exceed 76 characters. The other one you should consider after doing the other changes, because that may change its shape. You should also be able to surround the expression (but not the entire context expression, so the `open()` bit here) in parentheses.

On formatting, I recommend using `flake8` whose defaults are generally good. For modern python code I generally go for a longer line length, such as 100, but only where absolutely necessary.